### PR TITLE
feat(BAPP-700): Add talent protocol discount validator

### PR DESCRIPTION
**Pull request by stevieraykatz**
**What changed? Why?**

Integrate the talent protocol discount validator. 
Forked from #1089 

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/web/src/hooks/useAttestations.ts [hunk 1]
```diff
@@ -556,3 +559,37 @@ export function useDiscountCodeAttestations(code?: string) {
...
+    return {
+      address: discountValidatorAddress,
+      abi: TalentProtocolDiscountValidatorABI,
+      functionName: 'isValidDiscountRegistration',
+      args: [address, '0x0'],
+    };
+  }, [address, discountValidatorAddress]);
+
+  const { data: isValid, isLoading, error } = useReadContract({ ...readContractArgs, query: {} });
+  console.log();
```
Comment (kirkas) [inline apps/web/src/hooks/useAttestations.ts:582]: console log

File: apps/web/src/components/Basenames/RegistrationLearnMoreModal/index.tsx [hunk 2]
```diff
@@ -72,7 +73,7 @@ export default function RegistrationLearnMoreModal({
               <div className="flex flex-row items-center justify-start gap-2">
                 <ImageWithLoading
                   src={coinbaseVerification as StaticImageData}
-                  alt="criteria icon"
+                  alt=""
```
Comment (zencephalon) [inline apps/web/src/components/Basenames/RegistrationLearnMoreModal/index.tsx:76]: Hm, I think these should have alts, just the correct ones

Comment (stevieraykatz) [reply apps/web/src/components/Basenames/RegistrationLearnMoreModal/index.tsx]: Added descriptions of the various icons.

**Review by zencephalon — APPROVED** [review]
LFG

**Pull request merged**